docs/spec/openapi.yaml

```diff
@@ -0,0 +1,6930 @@
+        '403':
+          $ref: '#/components/responses/Forbidden'
+        '429':
+          $ref: '#/components/responses/TooManyRequests'
+      security:
+        - oauth_2_0: []
+  /browse/categories/{category_id}/playlists:
+    get:
+      tags:
+        - Playlists
+        - Categories
+      operationId: get-a-categories-playlists
+      summary: |
+        Get Category's Playlists
+      description: |
+        Get a list of Spotify playlists tagged with a particular category.
+      parameters:
+        - name: category_id
+          required: true
+          in: path
+          schema:
+            title: Category ID
+            description: |
+              The [Spotify category ID](/documentation/web-api/concepts/spotify-uris-ids) for the category.
+            example: dinner
+            type: string
+        - $ref: '#/components/parameters/QueryLimit'
+        - $ref: '#/components/parameters/QueryOffset'
+      responses:
+        '200':
+          $ref: '#/components/responses/PagedFeaturedPlaylists'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '403':
+          $ref: '#/components/responses/Forbidden'
+        '429':
+          $ref: '#/components/responses/TooManyRequests'
+      security:
+        - oauth_2_0: []
+  /playlists/{playlist_id}/images:
+    get:
+      tags:
+        - Playlists
+      operationId: get-playlist-cover
+      x-spotify-policy-list:
+        $ref: '#/components/x-spotify-policy/metadataWithMachineLearningPolicyList'
+      summary: |
+        Get Playlist Cover Image
+      description: |
+        Get the current image associated with a specific playlist.
+      parameters:
+        - $ref: '#/components/parameters/PathPlaylistId'
+      responses:
+        '200':
+          $ref: '#/components/responses/ArrayOfImages'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '403':
+          $ref: '#/components/responses/Forbidden'
+        '429':
+          $ref: '#/components/responses/TooManyRequests'
+      security:
+        - oauth_2_0: []
+    put:
+      tags:
+        - Playlists
+      operationId: upload-custom-playlist-cover
+      summary: |
+        Add Custom Playlist Cover Image
+      description: |
+        Replace the image used to represent a specific playlist.
+      parameters:
+        - $ref: '#/components/parameters/PathPlaylistId'
+      requestBody:
+        content:
+          image/jpeg:
+            schema:
+              example: /9j/2wCEABoZGSccJz4lJT5CLy8vQkc9Ozs9R0dHR0dHR0dHR0dHR0dHR0dHR0dHR0dHR0dHR0dHR0dHR0dHR0dHR0dHR0cBHCcnMyYzPSYmPUc9Mj1HR0dEREdHR0dHR0dHR0dHR0dHR0dHR0dHR0dHR0dHR0dHR0dHR0dHR0dHR0dHR0dHR//dAAQAAf/uAA5BZG9iZQBkwAAAAAH/wAARCAABAAEDACIAAREBAhEB/8QASwABAQAAAAAAAAAAAAAAAAAAAAYBAQAAAAAAAAAAAAAAAAAAAAAQAQAAAAAAAAAAAAAAAAAAAAARAQAAAAAAAAAAAAAAAAAAAAD/2gAMAwAAARECEQA/AJgAH//Z
+              format: byte
+              type: string
+              required: []
```

Comment:
I believe you encountered this error here:
```
OpenAPI definition has errors: 
attribute paths.'/playlists/{playlist_id}/images'(put).requestBody.content.'image/jpeg'.schema.required is not of type `array`
```
and you have fixed it by changing `required: true` to `required: []`. This might not be the ideal solution. 
According to the open API spec:

**Understanding the Error**

The error message you encountered is:
```
OpenAPI definition has errors: 
attribute paths.'/playlists/{playlist_id}/images'(put).requestBody.content.'image/jpeg'.schema.required is not of type `array`
```

This indicates that the required attribute within the schema for the image/jpeg content type is incorrectly defined. In OpenAPI, the required attribute within a schema is expected to be an array when defining required properties of an object. However, in your case, the schema is defining a primitive type (`string`), not an object with properties.

**Correct Usage of `required`**
1. For Object Schemas:

When defining an object with properties, required should be an array listing the names of the properties that are mandatory.

```yaml
type: object
properties:
  name:
    type: string
  age:
    type: integer
required:
  - name
```

2. For Request Bodies:

To indicate that the entire request body is required, use the required attribute at the requestBody level, not within the schema.

```yaml
requestBody:
  required: true
  content:
    image/jpeg:
      schema:
        type: string
        format: byte
        description: Base64 encoded JPEG image data, maximum payload size is 256 KB.
```

3. For Primitive Schemas:

The required attribute should not be used within the schema when the schema defines a primitive type like string, integer, etc.

**How to Fix Your OpenAPI Definition**

Remove the required Field from the Schema:

```yaml
put:
  tags:
    - Playlists
  operationId: upload-custom-playlist-cover
  summary: |
    Add Custom Playlist Cover Image
  description: |
    Replace the image used to represent a specific playlist.
  parameters:
    - $ref: '#/components/parameters/PathPlaylistId'
  requestBody:
    required: true  # Ensure the request body itself is required
    content:
      image/jpeg:
        schema:
          type: string
          format: byte
          example: /9j/2wCEABoZGSccJz4lJT5CLy8vQkc9Ozs9R0dHR0dHR0dHR0dHR0dHR0dHR0dHR0dHR0dHR0dHR0dHR0dHR0dHR0cBHCcnMyYzPSYmPUc9Mj1HR0dEREdHR0dHR0dHR0dHR0dHR0dHR0dHR0dHR0dHR0dHR0dHR0dHR0dHR0dHR0dHR//dAAQAAf/uAA5BZG9iZQBkwAAAAAH/wAARCAABAAEDACIAAREBAhEB/8QASwABAQAAAAAAAAAAAAAAAAAAAAYBAQAAAAAAAAAAAAAAAAAAAAAQAQAAAAAAAAAAAAAAAAAAAAARAQAAAAAAAAAAAAAAAAAAAAD/2gAMAwAAARECEQA/AJgAH//Z
          description: Base64 encoded JPEG image data, maximum payload size is 256 KB.
  responses:
    '202':
      description: Image uploaded
    '401':
      $ref: '#/components/responses/Unauthorized'
    '403':
      $ref: '#/components/responses/Forbidden'
    '429':
      $ref: '#/components/responses/TooManyRequests'
  security:
    - oauth_2_0:
        - ugc-image-upload
        - playlist-modify-public
        - playlist-modify-private
```

> Ensure required is Appropriately Placed: The required: true should be at the requestBody level to indicate that the request body is mandatory.